plugins/cpp_metrics/parser/src/cppmetricsparser.cpp

```diff
@@ -0,0 +1,89 @@
+#include <cppmetricsparser/cppmetricsparser.h>
+
+#include <model/cppastnodemetrics.h>
+#include <model/cppastnodemetrics-odb.hxx>
+#include <model/cppfilemetrics.h>
+#include <model/cppfilemetrics-odb.hxx>
+
+#include <model/cppastnode.h>
+#include <model/cppastnode-odb.hxx>
+
+#include <boost/filesystem.hpp>
+
+#include <util/logutil.h>
+#include <util/odbtransaction.h>
+
+#include <memory>
+
+namespace cc
+{
+namespace parser
+{
+
+CppMetricsParser::CppMetricsParser(ParserContext& ctx_): AbstractParser(ctx_)
+{
+}
+
+bool CppMetricsParser::accept(const std::string& path_)
+{
+  std::string ext = boost::filesystem::extension(path_);
+  return ext == ".dummy";
+}
+
+void CppMetricsParser::functionParameters()
+{
+  util::OdbTransaction {_ctx.db} ([&, this] {
+    for (const model::CppFunction func
+      : _ctx.db->query<model::CppFunction>())
+    {
+      model::CppAstNodeMetrics funcParams;
+      funcParams.astNodeId = func.astNodeId;
+      funcParams.type = model::CppAstNodeMetrics::Type::PARAMETER_COUNT;
+      funcParams.value = func.parameters.size();
```

Comment:
Lazy loading the function parameters for all functions creates a performance issue for medium size projects.
Please refactor this, e.g. use the existing `CppFunctionParamCount` ODB view from `cppfunction.h`.